### PR TITLE
Tests for - fix the case of per-function custom Role

### PR DESCRIPTION
**Pull request by Enase**
duplicate of https://github.com/HyperBrain/serverless-aws-alias/pull/145 with tests written for case when function has custom role name.

### CODE REVIEW

**Comment by Enase** [comment]
also, exception while test fixed for `updateAliasStack.test.js`

**Comment by Enase** [comment]
@HyperBrain I'm not sure about this fix, could you review?

**Comment by Enase** [comment]
Relates to https://github.com/serverless-heaven/serverless-aws-alias/pull/88 as well

BTW I use this fix in production, it's works fine.

**Comment by Enase** [comment]
@aleksdikanski could u review / approve?

File: lib/stackops/lambdaRole.js [hunk 1]
```diff
@@ -35,22 +35,26 @@ module.exports = function(currentTemplate, aliasStackTemplates, currentAliasStac
...
 	_.forEach(functions, func => {
-		func.Properties.Role = {
-			'Fn::GetAtt': [
-				roleLogicalId,
-				'Arn'
-			]
-		};
-		const dependencyIndex = _.indexOf(func.DependsOn, 'IamRoleLambdaExecution');
-		func.DependsOn[dependencyIndex] = roleLogicalId;
+		if (_.isEqual(func.Properties.Role, { 'Fn::GetAtt': ['IamRoleLambdaExecution', 'Arn' ]})) {
```
Comment (aleksdikanski) [inline lib/stackops/lambdaRole.js]: I would suggest to not use `if` in this instance but use `_.filter` to select the right elements. This goes more with the functional approach, e.g.,:
`_.forEach(_.filter(functions, func => _.isEqual(func.Properties.Role, { 'Fn::Gett': ... })), func => ...)`

Comment (Enase) [reply lib/stackops/lambdaRole.js]: got it

**Pull request merged**